docs/components/Autocomplete/Autocomplete.stories.mdx

```diff
@@ -38,6 +39,623 @@ the user types in the input. This async request should return a list of possible
+
+- `useKeyboardNavigation` is used to handle the default keyboard navigation of
+  the options
+- `MenuOption` is the default rendering of the options in the Autocomplete Menu.
+  The `UNSAFE_style` prop is used to add a custom styling to the options.
+- `isGroup` is a helper function that is used to determine if the option is a
+  group.
+- `isOptionSelected` is a helper function that is used to determine if the
+  option is selected.
+- `BaseMenuOption` is the base component for the options in the Autocomplete
+  Menu. This component provides the base styling and functionality for the
+  options such as the separators and the highlighting.
+
+<Tabs>
+<Tab label="Basic MenuOption Override">
+In this example we will override the default rendering of the Autocomplete Menu to add a custom styling to the options while keeping the default keyboard navigation.
+
+```tsx
+<Autocomplete
+  ...
+  customRenderMenu={({
+    MenuWrapper,
+    inputFocused,
+    onOptionSelect,
+    options,
+    selectedOption,
+    menuRef,
+  }) => {
+    const { highlightedIndex } = useKeyboardNavigation({
+      visible: inputFocused,
+            options,
+            onOptionSelect,
+            menuRef,
+          });
+
+    return (
+      <MenuWrapper visible={inputFocused}>
+        {options.map((option, index) => {
+          const optionStyle = getOptionStyling({
+            option,
+            highlightedIndex,
+            options,
+          });
+
+          return (
+            <MenuOption
+              key={index}
+              option={option}
+              addSeparators={isGroup(option)}
+              isHighlighted={index === highlightedIndex}
+              onOptionSelect={onOptionSelect}
+              isSelected={isOptionSelected(selectedOption, option)}
+              UNSAFE_style={optionStyle}
+            />
+          );
+        })}
+      </MenuWrapper>
+    );
+  }}
+/>
+```
+
+#### Full Example
+
+```tsx
+const sectionHeadingOptions = [
+  {
+    label: "Ships",
+    options: [
+      { value: 1, label: "Sulaco" },
+      { value: 2, label: "Nostromo" },
+      { value: 3, label: "Serenity" },
+      { value: 4, label: "Sleeper Service" },
+      { value: 5, label: "Enterprise" },
+      { value: 6, label: "Enterprise-D" },
+    ],
+  },
+  {
+    label: "Planets",
+    options: [
+      { value: 7, label: "Endor" },
+      { value: 8, label: "Vulcan" },
+      { value: 9, label: "Bespin" },
+      { value: 10, label: "Tatooine" },
+    ],
+  },
+];
+
+export const SimpleCustomMenu = () => {
+  const [value, setValue] = useState<Option | undefined>();
+
+  const planetSection = sectionHeadingOptions.find(
+    ({ label }) => label === "Planets"
+  )?.options;
+
+  return (
+    <Autocomplete
+      placeholder="Search for an option with section headings"
+      initialOptions={sectionHeadingOptions}
+      value={value}
+      onChange={newValue => setValue(newValue)}
+      customRenderMenu={({
```

Comment:
I see in the description you mention that one of the reasons for exposing the full menu is because the menu wouldn't be rendered if there are no options which is fair

looking at how much code this can result in even for a simple custom menu I do wonder if there isn't another way to unlock that behavior? maybe an `emptyOption` prop or something, and then only allow custom rendering of options

I still need to play around with this more, and maybe I'll change my mind on it but my initial thought is if people will find themselves needing access to the full menu where they need to rebuild a couple other layers - or if it's really just the options they need access to

Reply:
I will double check my notes (and update the PR) but I believe another reason has if the Input lost focused but you wanted to continue rendering the menu (such as with a footer action that is keyboard accessible). If I only allowed custom rendering of options then actions such as a footer button or a header action would eventually need to become a prop or they would need to add a fake option to determine if it is the header /footer